python/cuml/test/test_random_forest.py

```diff
@@ -167,23 +168,26 @@ def test_rf_classification(small_clf, datatype, split_algo,
                        n_bins=16, split_algo=split_algo, split_criterion=0,
```

Comment:
why do we default to the low n_bins still here? Maybe use the default?

Reply:
the reason was, the synthetic dataset size itself was only 100, so it was failing if bins were greater

Reply:
It fails after printing the reason saying the following
> The number of bins,`n_bins` can not be greater than the number of samples used for training.

In [this](https://github.com/rapidsai/cuml/blob/b30d527bd26635f6929984c4809445aa8ca39a57/python/cuml/ensemble/randomforest_common.pyx#L237) line.
Maybe I can then implicitly make `self.n_bins = max(self.n_rows, self.n_bins)` instead of raising a `ValueError`?

Reply:
> Maybe I can then implicitly make self.n_bins = min(self.n_rows, self.n_bins) instead of raising a ValueError?

I don't see an issue in doing it this way but I would feel much more comfortable if we do separate testing of this change. I recommend we keep the `n_bins` unchanged in the test code for CI to pass and revisit the issue later. Even with your proposed change, RF will operate on lower number of bins, so no real harm here. We can also think about increasing the size of dataset slightly to have more samples than bins.

Reply:
thanks vinay ✌🏻